openlibrary/core/ia.py

```diff
@@ -2,24 +2,239 @@
+            if isinstance(sorts, six.string_types):
+                sorts = sorts.split(',')
+            if isinstance(sorts, list):
+                # compare against field with +asc/desc suffix rm'd
+                # e.g: date, not date+asc or date+desc
+                return [
+                    sort.replace('+', ' ') for sort in sorts
+                    if sort.split('+')[0] in cls.VALID_SORTS
+                ]
+
+    @classmethod
+    def _expand_api_query(cls, query):
+        """Expands short / more easily cacheable preset queries
+        and fixes query to ensure only text archive.org items
+        are returned having openlibrary_work
+        :param str query: preset query or regular IA query
+        :rtype: str
+        """
+        # Expand preset queries
+        if query in cls.PRESET_QUERIES:
+            query = cls.PRESET_QUERIES[query]
+
+        q = 'mediatype:texts AND !noindex:* AND openlibrary_work:(*)'
+        # Add availability if none present (e.g. borrowable only)
+        if cls.AVAILABILITY_STATUS not in query:
+            q += ' AND %s:AVAILABLE' % cls.AVAILABILITY_STATUS
+        return q if not query else q + " AND " + query
+
+    @classmethod
+    def _clean_params(cls, q='', sorts=None, page=1, limit=None):
+        _limit = limit or cls.MAX_LIMIT
+        return {
+            'q': q,
+            'page': page,
+            'rows': min(_limit, cls.MAX_LIMIT),
+            'sort[]': sorts or '',
+            cls.RESPONSE_FIELDS: [
+                'identifier', cls.AVAILABILITY_STATUS, 'openlibrary_edition',
+                'openlibrary_work'
+            ],
+            'output': 'json'
+        }
+
+    @classmethod
+    def _compose_advancedsearch_url(cls, **params):
+        import urllib
+        return 'http://%s/advancedsearch.php?%s' % (
+            h.bookreader_host() or 'archive.org', urllib.urlencode(params, doseq=True))
+
+    @classmethod
+    def _compose_browsable_url(cls, query, sorts=None):
+        """If the client wants to explore this query in the browser, it will
+        have to be converted to user a human readable version of the
+        advancedsearch API
+        :param str query:
+        :param list sorts:
+        :rtype: str
+        :return: a url for humans to view this query on archive.org
+        """
+        _sort = sorts[0] if sorts else ''
+        if ' desc' in _sort:
+            _sort = '-' + _sort.split(' ')[0]
+        elif ' asc' in _sort:
+            _sort = _sort.split(' ')[0]
+        return '%s/search.php?query=%s&sort=%s' % (
+            h.ia_domain(), query, _sort)
+
+    @classmethod
+    def _request(cls, url):
+        """
+        Hits archive.org advancedsearch.php API, returns matching items
+        :param str url:
+        :rtype: dict
+        :return: the `response` content of the advancesearch
+        """
+        try:
+            import urllib2  # to port to requests
+            request = urllib2.Request(url)
```

Comment:
```suggestion
            request = six.moves.urllib.request.Request(url)
```
And lose the import line above.